jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/ResourceFactoryInternals.java

```diff
@@ -48,10 +49,17 @@
         CURRENT_WORKING_DIR = Path.of(System.getProperty("user.dir"));
 
         // The default resource factories
-        RESOURCE_FACTORIES.put("jar", new MountedPathResourceFactory());
+        MountedPathResourceFactory mountedPathResourceFactory = new MountedPathResourceFactory();
+        RESOURCE_FACTORIES.put("jar", mountedPathResourceFactory);
         PathResourceFactory pathResourceFactory = new PathResourceFactory();
         RESOURCE_FACTORIES.put("file", pathResourceFactory);
         RESOURCE_FACTORIES.put("jrt", pathResourceFactory);
+
+        URL url = ResourceFactoryInternals.class.getResource("/org/eclipse/jetty/version/build.properties");
```

Comment:
Is that code meant to discover the scheme of internal resources' URL? It's worth commenting a bit as this logic is mildly cryptic.

Also, are you sure `build.properties` is always going to be available? Can't you load the `ResourceFactoryInternals.class` file's bytes instead as you can be confident this is always available?

Reply:
Yeah, I'll add comments.

Oddly, on some environments (like GraalVM or Android) there are no class files to request as a resource.

Reply:
More comments added

Reply:
This check will not succeed for native-image environments unless you add a corresponding setting to allow access to to /org/eclipse/jetty/version/build.properties (e.g., via resource-config.json)

Reply:
Like the documentation says in the current HEAD of this branch, this is a best effort to discover schemes that are in use.

I'm aware that native-image needs more metadata to expose resources, but that's what your PR is for (and your PR is still relevant, this PR does not replace it)